node/connection_base.js

```diff
@@ -113,6 +114,13 @@ TChannelConnectionBase.prototype.onTimeoutCheck = function onTimeoutCheck() {
     }
     if (self.lastTimeoutTime) {
         self.timedOutEvent.emit(self);
+    } else if (!self.remoteName &&
+        self.timers.now() - self.startTime > self.channel.requestDefaults.timeout) {
+        self.timedOutEvent.emit(self, errors.TimeoutError({
```

Comment:
No it should be an error; this connection is dead; you must emit an error. Not getting an init req or init res breaks the connection; its not usable; its dead.